openquake/commonlib/valid.py

```diff
@@ -184,20 +184,34 @@ def __call__(self, value):
 
 name_with_dashes = Regex(r'^[a-zA-Z_][\w\-]*$')
 
-MAX_ID_LENGTH = 100
-
 
-def simple_id(value):
+class SimpleId(object):
     """
-    Check the source id; the only accepted chars are `a-zA-Z0-9_-`
+    Check if the given value is a valid ID.
+
+    :param length: maximum length of the ID
+    :param regex: accepted characters
     """
-    if len(value) > MAX_ID_LENGTH:
-        raise ValueError('The ID %r is longer than %d character' %
-                         (value, MAX_ID_LENGTH))
-    if re.match(r'^[\w_\-]+$', value):
-        return value
-    raise ValueError(
-        'Invalid ID %r: the only accepted chars are a-zA-Z0-9_-' % value)
+    def __init__(self, length, regex=r'^[\w_\-]+$'):
+        self.length = length
+        self.regex = regex
+        self.__name__ = 'SimpleId(%d, %s)' % (length, regex)
+
+    def __call__(self, value):
+        if len(value) > self.length:
+            raise ValueError('The ID %r is longer than %d character' %
+                             (value, self.length))
+        if re.match(self.regex, value):
+            return value
+        raise ValueError(
+            'Invalid ID %r: the only accepted chars are a-zA-Z0-9_-' % value)
+
+MAX_ID_LENGTH = 100
+ASSET_ID_LENGTH = 100
+
+simple_id = SimpleId(MAX_ID_LENGTH)
+asset_id = SimpleId(ASSET_ID_LENGTH)
```

Comment:
Is it possible that the asset IDs might include the "." and "|" characters as the source IDs do?